### PR TITLE
Add pypi tutorial

### PR DESCRIPTION
**Pull request by sed-i**
## Issue
Missing documentation on how to use the python plugin.

## Solution
Add tutorial for containerizing a pypi package.

### CODE REVIEW

**Comment by sed-i** [comment]
For the CLA, what do I put in this mandatory field?
` Please add the Canonical Project Manager or contact`

**Comment by simskij** [comment]
> For the CLA, what do I put in this mandatory field? ` Please add the Canonical Project Manager or contact`

You can put my name, I guess.

**Comment by tigarmo** [comment]
Hey thanks for this! I wonder why the cla check failed, isn't @sed-i in the Canonical org?

**Comment by sed-i** [comment]
> Hey thanks for this! I wonder why the cla check failed, isn't @sed-i in the Canonical org?

Sure am. Good question. Maybe because of the anonymized email address?
```
Checking the following users on Launchpad:
- 82407168+sed-i@users.noreply.github.com ✕ (has no Launchpad account)
- ```

File: docs/tutorials/code/pyfiglet/rockcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: pyfiglet
+base: ubuntu:22.04
+version: '0.7.6'  # Note: should match `pyfiglet` below
+summary: A rock for pyfiglet
```
Comment (tigarmo) [inline docs/tutorials/code/pyfiglet/rockcraft.yaml:4]:
```suggestion
summary: A ROCK for pyfiglet
```

**Review by tigarmo — COMMENTED** [review]
Thanks, this is nice! I left some suggestions and comments.

File: docs/tutorials/pypi-package.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    (_) |_  __      _____  _ __| | _____| |
+    | | __| \ \ /\ / / _ \| '__| |/ / __| |
+    | | |_   \ V  V / (_) | |  |   <\__ \_|
+    |_|\__|   \_/\_/ \___/|_|  |_|\_\___(_)
+
+
+Explore the running container
+-----------------------------
+
+To be able to poke around the container, you could add ``bash`` to
```
[the remark this answers is not in the document]
Comment (simskij) [reply docs/tutorials/pypi-package.rst:83]: The build succeeded, but it failed to let us drop into a shell in the container.

[the remark this answers is not in the document]
Comment (tigarmo) [reply docs/tutorials/pypi-package.rst:83]: Odd; which revision of `rockcraft` is this? It works for me (both via `--entrypoint` and via `exec bash`):

```bash
$ docker run --rm -it --entrypoint bash pyfiglet:latest
root@9a9477d7f9c3:/# 
exit
$ docker run --rm -it pyfiglet:latest exec bash
root@66547a144bed:~# 
exit

**Comment by tigarmo** [comment]
@sed-i @simskij the tutorials and how-tos in rockcraft are also ran as spread tests to ensure that they don't go stale; this entails moving the bash commands to a separate file and then referencing that in the .rst (see [this example](https://github.com/canonical/rockcraft/pull/241/files)).
We need to do this for this new tutorial too. Would you like to give it a try, or would you prefer if I do that (in which case I'll push on this branch if that's OK with you)?

**Comment by sed-i** [comment]
Hi @tigarmo,
Sorry for the delay.
I hope to get around to it later today or this week, but feel free to push to the branch regardless!

**Review by tigarmo — APPROVED** [review]

File: docs/tutorials/pypi-package.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    stage-packages:
+      - bash
+
+
+After repeating ``rockcraft pack`` and ``skopeo copy`` you should be able to
+override the entrypoint:
+
+.. code:: yaml
+
+    $ docker run --rm -it --entrypoint bash pyfiglet:0.7.6
```
[the remark this answers is not in the document]
Comment (tigarmo) [reply docs/tutorials/pypi-package.rst]: updated to this

File: docs/tutorials/pypi-package.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (tigarmo) [reply docs/tutorials/pypi-package.rst]: updated to `exec bash`

**Pull request merged**